app/models/foreman_azure_rm/azure_rm.rb

```diff
@@ -93,8 +93,32 @@ def test_connection(options = {})
       super(options)
     end
 
-    def new_vm(attr = {})
-      AzureRMCompute.new(sdk: sdk)
+    def new_vm(args = {})
+      if args.empty? || args[:image_id].nil?
+        AzureRMCompute.new(sdk: sdk)
+      else
+        opts = vm_instance_defaults.merge(args.to_h).deep_symbolize_keys
+        # convert rails nested_attributes into a plain hash
+        [:interfaces].each do |collection|
+          nested_args = opts.delete("#{collection}_attributes".to_sym)
+          opts[collection] = nested_attributes_for(collection, nested_args) if nested_args
+        end
+        opts.reject! { |k, v| v.nil? }
+
+        raw_vm = initialize_vm(location:        opts[:azure_vm][:location],
+                               vm_size:         opts[:vm_size],
+                               username:        opts[:username],
+                               password:        opts[:password],
+                               platform:        opts[:platform],
+                               ssh_key_data:    opts[:ssh_key_data],
+                               os_disk_caching: opts[:os_disk_caching],
+                               vhd_path:        opts[:image_id],
+                               premium_os_disk: opts[:premium_os_disk],
+                              )
+        vm = AzureRMCompute.new(azure_vm: raw_vm ,sdk: sdk)
+        vm.resource_group = opts[:resource_group]
```

Comment:
This works well in case of Compute Profile. However, it fails for normal host creation since there it is expecting `[:azure_vm][:resource_group]`. Playing around with this, I could understand that keeping the above hierarchy for **rg** would fail for host edit and keeping just [:resource_group] would fail for profile. Can we do something here where it can handle both the scenarios?
If still it errors out, how about removing the `Resource Group` field from profiles page?
@ShimShtein thoughts?